magma/compile.py

```diff
@@ -42,7 +42,15 @@ def _get_basename(basename):
     return basename
 
 
+class MagmaCompileException(Exception):
+    pass
+
+
 def compile(basename, main, output="coreir-verilog", **kwargs):
+    if not getattr(main, "is_definition", False):
+        raise MagmaCompileException(
+            f"Trying to compile empty definition {main}"
+        )
```

Comment:
```suggestion
    if not isdefinition(main):
        raise MagmaCompileException(
            f"Trying to compile empty definition {main}")
```

need to add `from magma.is_definition import isdefinition` after L9.